typings/glamorous.d.ts

```diff
@@ -40,6 +40,12 @@ interface ThemeProps {
 
 export class ThemeProvider extends React.Component<ThemeProps, any> { }
 
+export declare function withTheme
+  <P>(
+    component:
+      | React.ComponentClass<P>
```

Comment:
@debradley Looks good! Can you also add a test case in https://github.com/paypal/glamorous/blob/master/test/glamorous.test.tsx ?